data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/converter/RecordConverter.java

```diff
@@ -102,6 +105,7 @@ public void addToBuffer(final AcknowledgementSet acknowledgementSet,
         eventMetadata.setAttribute(DDB_STREAM_EVENT_NAME_METADATA_ATTRIBUTE, eventName);
         eventMetadata.setAttribute(EVENT_NAME_BULK_ACTION_METADATA_ATTRIBUTE, mapStreamEventNameToBulkAction(eventName));
         eventMetadata.setAttribute(EVENT_VERSION_FROM_TIMESTAMP, eventVersionNumber);
+        eventMetadata.setAttribute(DDB_STREAM_EVENT_USER_IDENTITY, userIdentity);
```

Comment:
It might make sense to only add this metadata attribute when it is a TTL delete. That way in the configuration later on for routes and things of that sort, users can just do a simple check on

```
getMetadata("ttl_delete") != null
```

rather than having to check explicitly for a true value with `getMetadata("ttl_delete") == true` . I am fine with it either way though.